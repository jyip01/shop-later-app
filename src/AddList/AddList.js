import React, { Component } from 'react'
import ShopLaterForm from '../ShopLaterForm/ShopLaterForm'
import ApiContext from '../ApiContext'
import config from '../config'
//import './AddList.css'

export default class AddList extends Component {
  static defaultProps = {
    history: {
      push: () => { }
    },
  }
  static contextType = ApiContext;

  handleSubmit = e => {
    e.preventDefault()
    const newList = {
      name: e.target['list-name'].value,
      content: e.target['list-content'].value,
      folderId: e.target['list-folder-id'].value,
      modified: new Date(),
    }
    fetch(`${config.API_ENDPOINT}/lists`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(newList),
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      })
      .then(list => {
        this.context.addList(list)
        this.props.history.push(`/folder/${list.folderId}`)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    const { folders=[] } = this.context
    return (
      <section className='AddList'>
        <h2>Create a list</h2>
        <ShopLaterForm onSubmit={this.handleSubmit}>
          <div className='field'>
            <label htmlFor='list-name-input'>
              Name
            </label>
            <input type='text' id='list-name-input' name='list-name' />
          </div>
          <div className='field'>
            <label htmlFor='list-content-input'>
              Content
            </label>
            <textarea id='list-content-input' name='list-content' />
          </div>
          <div className='field'>
            <label htmlFor='list-folder-select'>
              Folder
            </label>
            <select id='list-folder-select' name='list-folder-id'>
              <option value={null}>...</option>
              {folders.map(folder =>
                <option key={folder.id} value={folder.id}>
                  {folder.name}
                </option>
              )}
            </select>
          </div>
          <div className='buttons'>
            <button type='submit'>
              Add list
            </button>
          </div>
        </ShopLaterForm>
      </section>
    )
  }
}
