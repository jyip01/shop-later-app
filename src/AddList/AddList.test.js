import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddList from './AddList';

describe(`AddList component`, () => {
  const stubFolders = [
    {
      "id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Important"
    },
    {
      "id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Super"
    },
    {
      "id": "b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Spangley"
    }
  ]

  it('renders the complete form', () => {
    const wrapper = shallow(<AddList />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  // enzyme doesn't support React.createContext
  it.skip('renders the select options from folders', () => {
    const context = { folders: stubFolders }
    const select = shallow(<AddList />, context)
      .find('#list-folder-select')
    expect(toJson(select)).toMatchSnapshot()
  })
})
