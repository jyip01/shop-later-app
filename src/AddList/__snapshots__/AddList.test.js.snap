// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddList component renders the complete form 1`] = `
<section
  className="AddList"
>
  <h2>
    Create a list
  </h2>
  <ShopLaterForm
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label
        htmlFor="list-name-input"
      >
        Name
      </label>
      <input
        id="list-name-input"
        name="list-name"
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="list-content-input"
      >
        Content
      </label>
      <textarea
        id="list-content-input"
        name="list-content"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="list-folder-select"
      >
        Folder
      </label>
      <select
        id="list-folder-select"
        name="list-folder-id"
      >
        <option
          value={null}
        >
          ...
        </option>
      </select>
    </div>
    <div
      className="buttons"
    >
      <button
        type="submit"
      >
        Add list
      </button>
    </div>
  </ShopLaterForm>
</section>
`;
